packages/grid/src/mouse-handlers/GridLinkMouseHandler.ts

```diff
@@ -36,7 +36,7 @@ class GridLinkMouseHandler extends GridMouseHandler {
    * @param grid The grid
    * @returns A LinkInformation object if it exists and null otherwise
    */
-  static isHoveringLink(
+  static getHoveringLinkInfo(
```

Comment:
This probably shouldn't be static. It should be an instance method since there will be multiple instances of this handler at one time. While it probably shouldn't cause any issues with incorrect links (only 1 link can ever be hovered at a time), there's potentially some edge case where this being static causes some weird link issue when moving the cursor between grids, but I'm not sure

It's only called from within the class or a descendent class

Reply:
Right, but `IrisGridLinkMouseHandler` also calls the function to sense when to display the tooltip so I'm not sure how to handle that.

Reply:
For treeboxes, we calculate the size of the box in `GridMetricCalculator` and then stuff it in the `GridMetrics` as `visibleRowTreeBoxes`. I could imagine calculating the `visibleTokenBoxes` in `GridMetricCalculator` as well... though we do only _really_ need it for the hovered token... 

What you can do is refactor this logic into a method in `GridUtils` instead, e.g. 
```
export type TokenBox = BoxCoordinates & { token: Token };

getTokenFromXY(x: Coordinate, y: Coordinate, metrics: GridMetrics): TokenBox | null {
   ...
}
```
Then you can call it from `IrisGridTokenMouseHandler` and `GridTokenMouseHandler`, and they can each store their own instance of `currentTokenBox`. Alternatively, `GridTokenMouseHandler` stores it in it's instance, and `IrisGridTokenMouseHandler` iterates through `grid.mouseHandlers` to find the `GridTokenMouseHandler` instance.

Reply:
@emilyhuxng Actually thinking about this a bit more - it probably does make more sense to have this in `GridMetricCalculator`. So a method `getTokenBoxesForVisibleCell(row: VisibleIndex, column: VisibleIndex, state: GridMetricState): TokenBoxes` ... and then it only gets called from the mouse handlers for the cell that is hovered.

Boxes are relative to the cell... then we could memoize in GridMetricCalculator